index.js

```diff
@@ -11,63 +12,86 @@ const QUESTIONS = [{
         input: "[1,2,3,4,5]",
         output: "5"
     }]
-}];
+}]
 
 
 const SUBMISSION = [
 
 ]
 
-app.post('/signup', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
 
+app.post('/signup', (req, res)=>{
+    // Add logic to decode body
+    // body should have email and password
+    //check if user already exists
+    //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+    // return back 200 status code to the client
 
-  //Store email and password (as is for now) in the USERS array above (only if the user with the given email doesnt exist)
+    if(USERS.find(user=>user.username == req.body.username)){
 
+        return res.status(400).json({ error : 'User already exists'})
 
-  // return back 200 status code to the client
-  res.send('Hello World!')
-})
-
-app.post('/login', function(req, res) {
-  // Add logic to decode body
-  // body should have email and password
-
-  // Check if the user with the given email exists in the USERS array
-  // Also ensure that the password is the same
+    }else{
 
+    const user = {username: req.body.username, password: req.body.password}
+    USERS.push(user)
+    
+    res.status(200).json({ message: 'User registered successfully'})
 
-  // If the password is the same, return back 200 status code to the client
-  // Also send back a token (any random string will do for now)
-  // If the password is not the same, return back 401 status code to the client
-
-
-  res.send('Hello World from route 2!')
+    }
 })
 
-app.get('/questions', function(req, res) {
-
-  //return the user all the questions in the QUESTIONS array
-  res.send("Hello World from route 3!")
+app.post('/login', (req, res)=>{
+
+    // Add logic to decode body
+    // body should have email and password
+
+    // Check if the user with the given email exists in the USERS array
+    // Also ensure that the password is the same
+    // If the password is the same, return back 200 status code to the client
+    // Also send back a token (any random string will do for now)
+    // If the password is not the same, return back 401 status code to the client
+
+    const user = USERS.find(user => user.username == req.body.username)
+  if(!user){
+    return res.status(401).json({error: 'Invalid user_name'})
+  }
+  //check password
+  if(user.password !== req.body.password){
```

Comment:
Optional : its a goof practice to store passwords in hashed format. If Interested checkout `bcrypt`